templates/sls-app-template-java/src/main/java/com/slf/util/EnvironmentConfig.java

```diff
@@ -29,35 +29,38 @@
 
 	private static Properties props = new Properties();
 	private static String stage = null;
+	private static String functionName = null;
 
-	public EnvironmentConfig (Context context) throws Exception {
-		String fnName = context.getFunctionName();
-		String fnNameNoEnv = null;
-		String fnNoPrefix = null;
-
-		if(null != fnName) {
-			int lastIndx = fnName.lastIndexOf("-");
-			fnNoPrefix = fnName.substring(lastIndx+1);
-			fnNameNoEnv = fnName.substring(0, lastIndx);
-			int preIndx = fnNameNoEnv.lastIndexOf("-");
-			if(preIndx < 0) { stage = fnNameNoEnv; }
-			else {
-				stage = fnNameNoEnv.substring(preIndx+1);
-			}
+	public EnvironmentConfig(Context context) throws Exception {
+		functionName = context.getFunctionName();		
 
+		if (null != functionName) {
+			int lastIndx = functionName.lastIndexOf("-");
+			stage = functionName.substring(lastIndx + 1);			
 		}
 
-		if(stage.isEmpty()) {
+		if (stage.isEmpty()) {
 			throw new BadRequestException("Invalid Stage. Can't load ENV configurations");
 		}
-
-		String configFile = "/functions/"+fnNoPrefix+"/"+stage+".properties";
-		logger.info("Loading configuration file for env..:"+configFile);
+		
+	}
+	
+	public void loadConfig() throws Exception {
+		StackTraceElement ste = Thread.currentThread().getStackTrace()[2];
```

Comment:
Is this convoluted because of the structure of the template?

Reply:
Its dependent on the function(lambda)name.